src/main/java/com/mparticle/kits/OneTrustKit.java

```diff
@@ -12,16 +32,90 @@
     private final static String MP_MOBILE_CONSENT_GROUPS = "mobileConsentGroups";
     private final static String ONETRUST_PREFS = "OT_mP_Mapping";
     private Context m_context;
-    
+
+    private BroadcastReceiver categoryReceiver;
+
     @Override
-    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, final Context context) {
         // Retrieve mParticle --> OneTrust mapping values
         String mobileMappingValues = settings.get(MP_MOBILE_CONSENT_GROUPS);
+
+        final Map<String, String> consentMapping = new HashMap<String, String>();
+
+        if (mobileMappingValues != null) {
+            try {
+                // Convert mobileMappingValues and extract new consentMapping
+                // value -> OneTrust Cookie Value
+                // map -> mParticle Consent Purpose
+                JSONArray consentJSONArray = new JSONArray(mobileMappingValues);
+                for (int i = 0; i < consentJSONArray.length(); i++) {
+                    JSONObject consentJSONObject = consentJSONArray.optJSONObject(i);
+
+                    consentMapping.put(consentJSONObject.getString("value"), consentJSONObject.getString("map"));
+                }
+
+            } catch (JSONException jsonException) {
+                Logger.warning(jsonException, "MP OT JSON ERROR");
+            }
+        }
+
+        // Listen for One Trust Consent changes and create ne mPartcle Consent
+        // Changes based on these events
+        // intent.getAction -> OT cookie name
+        // intent.getIntExtras(OTBroadcastServiceKeys.EVENT_STATUS) -> (bool-int) consented?
+        categoryReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                Log.i("BroadcastService", "MP OT Intent name: " + intent.getAction() +
+                        " status = " + intent.getIntExtra(OTBroadcastServiceKeys.EVENT_STATUS, -1));
+
+                OneTrustKit.this.createConsentEvent(intent.getAction(), intent.getIntExtra(OTBroadcastServiceKeys.EVENT_STATUS, -1));
+            }
+        };
+
+        for(final String consentElement: consentMapping.keySet()) {
+            // Register receiver from above based on cookie value dispatched by OneTrust SDK
+            context.registerReceiver(categoryReceiver, new IntentFilter(consentElement));
+
+            // Fetch Consent Status from OneTrust based on cookie value
+            new OTPublishersHeadlessSDK(context).getConsentStatusForGroupId(consentElement);
+
+            // Dispatch creation of initial consent state till after init is done
+            new Handler().post(new Runnable() {
+                @Override
+                public void run() {
+                    OneTrustKit.this.createConsentEvent(consentMapping.get(consentElement), 0);
```

Comment:
did you accidentally leave in the hardcoded `0`? It looks like the return value from line 81 should be captured as a variable and passed in here